src/scatter.ts

```diff
@@ -0,0 +1,208 @@
+import * as d3 from "d3";
+import { Unit } from ".";
+import "./styles.css"
+
+/**
+ * Creates a scatter plot based on different unit specifications selected by the user.
+ * Uses a dropdown list per axis to pick attributes for comparison.
+ */
+export async function createScatter(data: Unit[]) {
+
+    // Set the dimensions and margins of the graph
+    var margin = {top: 10, right: 30, bottom: 20, left: 60},
+    width = 460 - margin.left - margin.right,
+    height = 400 - margin.top - margin.bottom;
+    d3.select("#scatter").selectAll("svg").remove();
+
+    // Append the svg object to the body of the page
+    const svg = d3.select("#scatter")
+    .append("svg")
+    .attr("width", width + margin.left + margin.right)
+    .attr("height", height + margin.top + margin.bottom)
+    .append("g")
+    .attr("transform",
+        "translate(" + margin.left + "," + margin.top + ")");
+
+    d3.select("#dropdown_container_x").selectAll("select").remove();
+    d3.select("#dropdown_container_y").selectAll("select").remove();
+
+    // Clickable options in the dropdown list
+    const dropdown_list = [
+        "weapon damage", "weapon max range", "movement speed", 
+        "gold costs", "attack duration", "armor",
+        "modifier max value"
+    ];
+    
+    createDropDown(dropdown_list, "x_selection", "x");
+    createDropDown(dropdown_list.reverse(), "y_selection", "y");
+    
+    d3.select('#x_selection')
+    .on('change', function() {
+        updateScatterPlot(data, width, height, margin);       
+    });
+
+    d3.select('#y_selection')
+    .on('change', function() {
+        updateScatterPlot(data, width, height, margin);
+    });
+
+    updateScatterPlot(data, width, height, margin);
+
+    return svg.node();
+}
+
+/**
+ * Returns specific data to plot based on the selected option for a specific axis
+ * @param  {Unit}   unit           unit to return data from
+ * @param  {string} selection_name option selected in dropdown list
+ * @return {number}             numerical value to plot in the scatterplot
+ */
+function returnData(unit: Unit, selection_name: string, debug = false): number {
```

Comment:
Maybe we should share the definitions for the variables which can be plotted to be same as those which can be filtered? If you look at the code for the filters, I define each as a tuple between the name of the field, and a function which returns the field when given the unit. Could you use the same?

Reply:
I think we will improve it in the new PR to help the work on the interaction